course3/UniquePtr/my_unique_ptr.h

```diff
@@ -0,0 +1,221 @@
+#include <iostream>
+/*
+re-implemention of http://en.cppreference.com/w/cpp/memory/unique_ptr
+*/
+
+template<class T>
+class MyUniquePtr
+{
+public:
+    MyUniquePtr() = default;
+    
+    ~MyUniquePtr(){}
+    
+    MyUniquePtr(T*&& ptr) : mData(ptr)
```

Comment:
This makes no sense. Passing simple type (pointer) by rvalue ref doesn't really man much.
And look at the logic: you are storing a pointer to T, and immediately after you delete it. So thus constructed object holds a pointer to deleted object.

Please, write unit test, and run them via valgrind. I expect an avalanche of memory errors and leaks.